src/Microsoft.Azure.ServiceBus/Management/ManagementClient.cs

```diff
@@ -1,101 +1,95 @@
         {
-            this.csBuilder = connectionStringBuilder;
-            this.ServiceBusConnection = new ServiceBusConnection(connectionStringBuilder);
-            this.ServiceBusConnection.RetryPolicy = this.RetryPolicy;
-            this.ownsConnection = true;
-            this.httpClient = new HttpClient();
-            this.port = GetPort(connectionStringBuilder.Endpoint);
         }
 
-        public ManagementClient(string connectionString, RetryPolicy retryPolicy = null)
-            : this(new ServiceBusConnectionStringBuilder(connectionString), retryPolicy)
+        public ManagementClient(string endpoint, ITokenProvider tokenProvider, RetryPolicy retryPolicy = default )
+            : this(new ServiceBusConnectionStringBuilder { Endpoint = endpoint}, tokenProvider, retryPolicy)
         {
         }
 
-        public ManagementClient(string endpoint, ITokenProvider tokenProvider, RetryPolicy retryPolicy = null)
-            : this(new ServiceBusConnectionStringBuilder() { Endpoint = endpoint}, retryPolicy)
+        public ManagementClient(ServiceBusConnectionStringBuilder connectionStringBuilder, ITokenProvider tokenProvider = default, RetryPolicy retryPolicy = default)
         {
-            this.ServiceBusConnection.TokenProvider = tokenProvider;
+            this.httpClient = new HttpClient();
+            this.endpointFQDN = connectionStringBuilder.Endpoint;
+            this.retryPolicy = retryPolicy ?? RetryPolicy.Default;
+            this.tokenProvider = tokenProvider ?? CreateTokenProvider(connectionStringBuilder);
+            this.operationTimeout = Constants.DefaultOperationTimeout;
+            this.port = GetPort(connectionStringBuilder.Endpoint);
         }
 
-        public override ServiceBusConnection ServiceBusConnection { get; }
-
-        public override TimeSpan OperationTimeout
+        private static ITokenProvider CreateTokenProvider(ServiceBusConnectionStringBuilder builder)
         {
-            get => this.ServiceBusConnection.OperationTimeout;
-            set => this.ServiceBusConnection.OperationTimeout = value;
-        }
-
-        public override string Path => this.ServiceBusConnection.Endpoint.AbsoluteUri;
+            if (builder.SasToken != null)
+            {
+                return new SharedAccessSignatureTokenProvider(builder.SasToken);
+            }
 
-        public override IList<ServiceBusPlugin> RegisteredPlugins => null;
+            if (builder.SasKeyName != null || builder.SasKey != null)
+            {
+                return new SharedAccessSignatureTokenProvider(builder.SasKeyName, builder.SasKey);
+            }
 
-        public override void RegisterPlugin(ServiceBusPlugin serviceBusPlugin)
-        {
-            throw new NotImplementedException($"{nameof(ManagementClient)} doesn't support plugins");
+            // TODO: is this even possible?
+            throw new Exception("Could not create token provider from the provided connection string.");
```

Comment:
Yes. When they purposefully pass null when using ITokenProvider and don't have SAS information in the CsBuilder. 
Throw `ArgumentException` instead here.

Reply:
`ArgumentException` should indicate a single parameter issue. In this case it's 2 things: 1) token provider is null _and_ connection string doesn't include what we need. I still think a general exception is OK, just with a different error message (feel free to suggest).

Reply:
Maybe remove the TODO then? :)

---
In reply to: [196201085](https://github.com/Azure/azure-service-bus-dotnet/pull/493#discussion_r196201085) [](ancestors = 196201085)

Reply:
Will be in PR #495